server/sessionruntime/service.go

```
@@ -84,6 +87,62 @@ func NewService(persistenceRoot string, metadataStore *metadata.Store, authManag
+}
+
+type recoveredWarningEntry struct {
+	Visibility transcript.EntryVisibility `json:"visibility,omitempty"`
+	Role       string                     `json:"role"`
+	Text       string                     `json:"text"`
+}
+
+func (s *Service) appendRecoveredWarningIfNeeded(store *session.Store) error {
+	warning, ok, err := s.generatedRecoveredWarning()
+	if err != nil {
+		return nil
```

golangci-lint (2.12.1)

[error] error is not nil (line 113) but it returns nil

(nilerr)